inlong-tubemq/tubemq-manager/src/main/java/org/apache/inlong/tubemq/manager/service/TopicServiceImpl.java

```diff
@@ -111,25 +117,143 @@ public TubeMQResult queryGroupExist(QueryConsumerGroupReq req) {
+                || topicName.length() > MAX_TOPIC_NAME_LENGTH) {
+            log.error("Invalid topicName: ClusterId = {}, TopicName = {}", clusterId, topicName);
+            throw new IllegalArgumentException("Invalid topicName.");
+        }
+    }
+
+    /**
+     * Builds the URL for requesting topic view information.
+     *
+     * @param masterNode The MasterEntry representing the TubeMQ master node.
+     * @param topicName  The name of the topic.
+     * @param clusterId  The ID of the TubeMQ cluster.
+     * @return           The constructed URL.
+     * @throws IllegalArgumentException If the URL is invalid.
+     */
+    private String buildTopicViewURL(MasterEntry masterNode, String topicName, Long clusterId) {
+        String url = TubeConst.SCHEMA + masterNode.getIp() + ":" + masterNode.getWebPort() + TubeConst.TOPIC_VIEW;
+        if (!isValidURL(url)) {
+            log.error("Invalid URL: ClusterId = {}, URL = {}", clusterId, url);
+            throw new IllegalArgumentException("Invalid URL.");
+        }
+        return url + TubeConst.TOPIC_NAME + topicName;
+    }
+
+    /**
+     * Parses the response to obtain a TopicView object.
+     *
+     * @param response The HTTP response containing topic view information.
+     * @return         The parsed TopicView object.
+     * @throws Exception If an exception occurs during parsing.
+     */
+    private TopicView parseTopicViewResponse(CloseableHttpResponse response) throws Exception {
+        return gson.fromJson(new InputStreamReader(response.getEntity().getContent(), StandardCharsets.UTF_8),
+                TopicView.class);
+    }
+
+    /**
+     * Handles exceptions that occur during the request.
+     *
+     * @param clusterId The ID of the TubeMQ cluster.
+     * @param topicName The name of the topic.
+     * @param ex        The exception that occurred.
+     */
+    private void handleRequestException(Long clusterId, String topicName, Exception ex) {
+        log.error("Exception caught while requesting topic view: ClusterId = {}, TopicName = {}", clusterId, topicName,
+                ex);
+    }
+
+    /**
+     * Checks if the input string contains dangerous characters or keywords that may pose security risks,
+     * such as those commonly associated with SSRF attacks.
+     *
+     * @param input The input string to be checked for dangerous characters or keywords.
+     * @return True if the input contains dangerous characters or keywords, otherwise false.
+     */
+    private boolean containsDangerousChars(String input) {
+        input = input.toLowerCase();
+        String[] dangerousKeywords =
+                {"exec", "system", "cmd", "shell", "php", "perl", "python", "ruby", "javascript", "java"};
+        // Prevent SSRF attacks by checking for "://"
+        if (input.contains("://")) {
+            return true;
+        }
+        // Check for other possible dangerous characters or keywords
+        for (String keyword : dangerousKeywords) {
```

Comment:
Set::contains

Reply:
Thanks for the suggestion, changes have been made